src/live/handleAlbumChat.js

```diff
@@ -1,11 +1,30 @@
-function handleAlbumChat(io, socket, data) {
-  const { message, userId, albumId } = data;
+const ChatMessage = require('../models/ChatMessage');
 
-  // Create a chat room name for the album chat
-  const chatRoomName = `chat:album:${albumId}`;
+async function handleAlbumChat(io, socket, data) {
+  try {
+    const { albumId } = data;
 
-  // Send the message to the room
-  socket.to(chatRoomName).emit('chat:album', { message, userId, albumId });
+    // Create a chat room name for the album chat
+    const chatRoomName = `chat:album:${albumId}`;
+
+    const msg = await ChatMessage.create(data);
+    await msg.populate('user', 'name');
+    console.log('Chat message saved:', msg);
+
+    // Check and delete old messages if needed
+    const messagesCount = await ChatMessage.countDocuments({ albumId });
+    if (messagesCount > 50) {
+      await ChatMessage.findOneAndDelete(
+        { albumId },
+        { sort: { timestamp: 1 } }
+      );
+    }
+
+    // Broadcast the message to all connected users in the album chat room
+    io.to(chatRoomName).emit('chat:album', data);
```

Comment:
You're sending back the original `data` object that the frontend sent:

```

{
  "message": "Hi",
  "albumId": "6r1lh7fHMB499vGKtIyJLy",
  "user": "65297a45ccf10113ea16cf7e"
}
```

This won't have the populated user name or any other info that we could get from the `ChatMessage` that we created on line 10. It looks like you changed this from returning the `msg` variable to returning `data` in the last commit: 48203fe

Could you help me understand if there was something missing from `msg` that led you to change that?

Reply:
I probably didn’t understand you correctly 3 days ago, I thought I needed to make these changes, I’m returning it back.